clusterloader: add a new flag delete-automanaged-namespaces

yuanchen8911 opened the pull request:
clusterloader automatically creates namespaces for testing and deletes the namespaces and  all resources in the namespaces after the test exits (successfully or failed).  We propose adding a flag _delete-auto-namespaces_ to specify whether or not delete the namespace after a test.

wojtek-t commented on clusterloader2/pkg/test/simple_test_executor.go:
```diff
@@ -372,6 +372,9 @@ func isErrsCritical(*errors.ErrorList) bool {
 }
 
 func cleanupResources(ctx Context) {
+	if  !ctx.GetClusterFramework().GetClusterConfig().DeleteAutomanagedNamespaces {
```
nit: I can imagine that we will be extending cleanupResources to cleanup other stuff in the future.

So I would prefer being more explicit and move it below in a form:
```
if ctx.....DeleteAutomanagedNamespaces {
  if errList := ctx.....DeleteAutomanagedNamespaces(); ...
    // as is now
   }
}

yuanchen8911 replied:
sounds good.

yuanchen8911 replied:
made the change.

yuanchen8911 commented:
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
I already signed a CLA and set the email address.

> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
No, I was signed as an individual contributor.

wojtek-t commented:
@yuanchen8911 - please squash the commits and sign the CLA and I will approve.

yuanchen8911 commented:
> @yuanchen8911 - please squash the commits and sign the CLA and I will approve.
@wojtek-t 

1. squashed and rebased.  

2.  CLA form attached  
[CNCF_Individual_Contributor_License_Agreement-3.pdf](https://github.com/kubernetes/perf-tests/files/4458490/CNCF_Individual_Contributor_License_Agreement-3.pdf)

yuanchen8911 commented:
/assign @krzysied

yuanchen8911 commented:
/assign @krzysied

yuanchen8911 commented:
/assign @wojtek-t

wojtek-t commented:
@yuanchen8911 - please rebase properly - it contains commits that are already merged.

wojtek-t commented on clusterloader2/pkg/config/cluster.go:
```diff
@@ -42,6 +42,7 @@ type ClusterConfig struct {
 	MasterName                 string
 	KubemarkRootKubeConfigPath string
 	DeleteStaleNamespaces      bool
+	DeleteAutomanagedNamespaces bool
```
I think gofmt will fail with it - please run gofmt.

yuanchen8911 replied:
Checked the format

wojtek-t commented on clusterloader2/pkg/test/simple_test_executor.go:
```diff
@@ -374,9 +374,11 @@ func isErrsCritical(*errors.ErrorList) bool {
+	    if errList := ctx.GetClusterFramework().DeleteAutomanagedNamespaces(); !errList.IsEmpty() {
 		klog.Errorf("Resource cleanup error: %v", errList.String())
 		return
+            }
```
Same here - please run gofmt.

yuanchen8911 replied:
Checked the format of all 3 files.

yuanchen8911 commented:
> @yuanchen8911 - please rebase properly - it contains commits that are already merged.

@wojtek-t  I rebased from master. Hope it works now. Thanks.

wojtek-t commented:
/lgtm
/approve
